packages/dd-trace/test/setup/core.js

```diff
@@ -48,3 +49,35 @@ if (/^v\d+\.x$/.test(process.env.GITHUB_BASE_REF || '')) {
   process.env.DD_INJECTION_ENABLED = 'true'
   process.env.DD_INJECT_FORCE = 'true'
 }
+
+// TODO(bengl): remove this block once we can properly support Node.js 24 without it
+if (NODE_MAJOR >= 24 && !process.env.OPTIONS_OVERRIDE) {
+  const childProcess = require('child_process')
+  const { exec, fork } = childProcess
+
+  childProcess.exec = function (...args) {
+    const opts = args[1]
+    if (opts) {
+      if (opts?.env?.NODE_OPTIONS && !opts.env.NODE_OPTIONS.includes('--no-async-context-frame')) {
+        opts.env.NODE_OPTIONS += ' --no-async-context-frame'
+      } else {
+        opts.env ||= {}
+        opts.env.NODE_OPTIONS = '--no-async-context-frame'
+      }
+    }
+    return exec.apply(this, args)
+  }
+
+  childProcess.fork = function (...args) {
+    const opts = args[1]
+    if (opts) {
+      if (opts?.env?.NODE_OPTIONS && !opts.env.NODE_OPTIONS.includes('--no-async-context-frame')) {
```

Comment:
I believe the second condition should actually be an inner if and it would just skip the addition. Otherwise the presence of the flag would remove all other NODE_OPTIONS potentially also being set.

Reply:
I think you're right. While this is only in test code where we control NODE_OPTIONS, we don't necessarily know _where_ we might want to add other things, so this is worth fixing. I'll do that in a follow-up PR in a few min. 👍